question28/js/modules/validations.js

```diff
@@ -51,3 +51,23 @@ export const validateInputValue = (e) => {
   const result = validationInfo[targetForm.name].validation(value);
   setValidationEvents(result,targetForm);
 } 
+
+export const validateInputValueForButton = (targetForm) => {
+  const value = targetForm.value.trim();
+  const result = validationInfo[targetForm.name].validation(value);
+  setValidationEvents(result,targetForm);
+} 
+
+export const validationForTargetForm = (e) => {
+  if (e.relatedTarget === e.target.parentNode.querySelector('input')) return;
+  validateInputValueForButton(e.target.parentNode.querySelector('[data-input="password"]')); 
+}
+
+export const resetValidation = (e) => {
+  const errorMessage = e.target.parentNode.querySelector('[data-error="error"]');
+  const inputOfPassword = e.target.parentNode.querySelector('[data-input="password"]');
+  errorMessage.textContent = ""; 
+  inputOfPassword.classList.remove("invalid");
+  inputOfPassword.classList.remove("valid");
+  inputOfPassword.classList.remove("invalid_mismatch");
+}
```

Comment:
nits
モジュール化するアイディアとても良いと思いました👏
classList.removeはまとめた方がスマートでもっと良くなると思いました!
[ref](https://developer.mozilla.org/ja/docs/Web/API/Element/classList)
```suggestion
export const resetValidation = (e) => {
  const errorMessage = e.target.parentNode.querySelector('[data-error="error"]');
  const inputOfPassword = e.target.parentNode.querySelector('[data-input="password"]');
  errorMessage.textContent = ""; 
  inputOfPassword.classList.remove("invalid","valid","invalid_mismatch");
}
```

Reply:
ご指摘ありがとうございます！
短縮できるんですね、とてもスッキリしました:laughing:  👏
大変勉強になりました! ご指摘の通り修正させていただきましたのでご確認いただけますと幸いです!

fixed commit : 
https://github.com/csakyo/homework/pull/46/commits/dfc75130a7ad41363722226e228f6e70496c0414